Library/Homebrew/rubocops/cask/ast/stanza.rb

```diff
@@ -12,23 +12,34 @@ module AST
       class Stanza
         extend Forwardable
 
+        sig {
+          params(
+            method_node:  T.any(RuboCop::AST::SendNode, RuboCop::AST::BlockNode),
+            all_comments: T::Array[T.any(String, Parser::Source::Comment)],
+          ).void
+        }
         def initialize(method_node, all_comments)
-          @method_node = method_node
-          @all_comments = all_comments
+          @method_node = T.let(method_node, T.any(RuboCop::AST::SendNode, RuboCop::AST::BlockNode))
```

Comment:
So the comment at the top of this `class` says "send/block node", but I'm seeing cop errors with:

```
An error occurred while Cask/UrlLegacyCommaSeparators cop was inspecting /opt/homebrew/Library/Homebrew/test/support/fixtures/cask/Casks/multiple-versions.rb:1:0.
Parameter 'method_node': Expected type T.any(RuboCop::AST::BlockNode, RuboCop::AST::SendNode), got type RuboCop::AST::AsgnNode with value s(:lvasgn, :platform,
  s(:...
        s(:str, "darwin")))))
Caller: /opt/homebrew/Library/Homebrew/rubocops/cask/ast/cask_block.rb:71
Definition: /opt/homebrew/Library/Homebrew/rubocops/cask/ast/stanza.rb:21 (RuboCop::Cask::AST::Stanza#initialize)
```

Reply:
The comment at the top is six years old. Even if correct at the time, it's likely to drift without enforcement of the contract (which is what sorbet is for). You could try adding adding an add'l `select` at the caller and see if that maintains compatibility, or just update the sig/comment to allow the add'l node type.

(Also, rather than union a bunch of node types, I'd probably just use the `RuboCop::AST::Node` superclass, with type guards within the method definition when i need a more specific API. A lot of that is a judgement call based on the resulting code though.)

Reply:
Thanks! I tried `RuboCop::AST::Node` first but then got way more errors in different places because other places were expecting `BlockNode` and I didn't want to `T.cast` a bunch. Let me try again!